packages/twenty-front/src/modules/apollo/hooks/useApolloFactory.ts

```diff
@@ -105,5 +107,11 @@ export const useApolloFactory = (options: Partial<Options<any>> = {}) => {
     }
   }, [tokenPair]);
 
+  useUpdateEffect(() => {
+    if (isDefined(apolloRef.current)) {
+      apolloRef.current.updateWorkspaceMember(currentWorkspaceMember);
+    }
+  }, [currentWorkspaceMember]);
+
```

Comment:
One thing we'll need to do in a followup PR is also refetch the ObjectMetadataItems query when the locale is changed (can be done directly in the called function, no need for an effect there).